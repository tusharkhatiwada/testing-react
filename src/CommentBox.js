import React from "react";

export default class CommentBox extends React.Component {
    state = {
        comment: ""
    };
    handleFormSubmit = event => {
        event.preventDefault();
        this.setState({ comment: "" });
    };
    render() {
        return (
            <div>
                <form onSubmit={this.handleFormSubmit}>
                    <div>
                        <textarea
                            value={this.state.comment}
                            onChange={event => this.setState({ comment: event.target.value })}
                        />
                    </div>
                    <button>Submit</button>
                </form>
            </div>
        );
    }
}
