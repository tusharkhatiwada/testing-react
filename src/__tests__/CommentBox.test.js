import React from "react";
// mount is imported from Full DOM Rendering
import { mount } from "enzyme";

// import for snapshot testing. npm install --save react-test-renderer
import renderer from "react-test-renderer";

import CommentBox from "../CommentBox";

describe("CommentBox", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(<CommentBox />);
    });

    afterEach(() => {
        // unmount after the test case for performance
        wrapper.unmount();
    });

    it("has a textarea and a button", () => {
        expect(wrapper.find("textarea").length).toEqual(1);
        expect(wrapper.find("button").length).toEqual(2);
    });

    it("has a textarea and receives a value", () => {
        // simulates the change event of textarea
        wrapper.find("textarea").simulate("change", {
            target: {
                value: "New Comment"
            }
        });

        // update the component
        wrapper.update();

        // expectation
        expect(wrapper.find("textarea").prop("value")).toEqual("New Comment");
    });

    it("clears the textarea when form is submit", () => {
        wrapper.find("textarea").simulate("change", {
            target: {
                value: "New Comment"
            }
        });

        wrapper.update();

        // simulates form sumbit event
        wrapper.find("form").simulate("submit");

        wrapper.update();

        expect(wrapper.find("textarea").prop("value")).toEqual("");
    });

    it("renders correctly", () => {
        const tree = renderer.create(<CommentBox />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
