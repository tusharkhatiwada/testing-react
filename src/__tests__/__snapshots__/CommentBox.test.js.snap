// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentBox renders correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div>
      <textarea
        onChange={[Function]}
        value=""
      />
    </div>
    <button>
      Submit
    </button>
    <button>
      Cancel
    </button>
  </form>
</div>
`;
